### ISSUE TITLE
Assertion `!bail_error || match' failed to hold. Fatal register mismatch (ticks/rec:7515/7515)

### ISSUE THREAD

**Issue by ehsan** [mozilla/rr#1605]
I am seeing various issues with rr, and this one I seem to be able to reproduce on two recordings.  I have recorded a Firefox crash in the parent process.  Here is what happens when I replay:

```
$ rr replay -p firefox
GNU gdb (Ubuntu 7.7.1-0ubuntu5~14.04.2) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/ehsan/.rr/firefox-1/mmap_1_hardlink_firefox...done.
Really redefine built-in command "checkpoint"? (y or n) [answered Y; input not from terminal]
Really redefine built-in command "checkpoint"? (y or n) [answered Y; input not from terminal]
Really redefine built-in command "restart"? (y or n) [answered Y; input not from terminal]
/proc/16893/fd/4:61: Error in sourced command file:
Undefined maintenance set command: "target-async 0".  Try "help maintenance set".
Remote debugging using 127.0.0.1:16894

--------------------------------------------------
 ---> Reached target process 16733 at event 15.
--------------------------------------------------
warning: limiting remote suggested packet size (206696 bytes) to 16384
Reading symbols from /lib64/ld-linux-x86-64.so.2...Reading symbols from /usr/lib/debug//lib/x86_64-linux-gnu/ld-2.19.so...done.
done.
Loaded symbols for /lib64/ld-linux-x86-64.so.2
0x00007fb3c3a742d0 in _start () from /lib64/ld-linux-x86-64.so.2
(rr) c
Continuing.
Loading JavaScript value pretty-printers; see js/src/gdb/README.
If they cause trouble, type: disable pretty-printer .* SpiderMonkey
[New Thread 16733.16737]
[ERROR /home/ehsan/src/rr/src/Registers.cc:332:maybe_print_reg_mismatch() errno: 0 'Success']
 -> eflags 0x6 != 0 (replaying vs. recorded)
[FATAL /home/ehsan/src/rr/src/Registers.cc:450:compare_register_files() errno: 0 'Success']
 (task 16914 (rec:16748) at time 4025)
 ->  Assertion `!bail_error || match' failed to hold. Fatal register mismatch (ticks/rec:7515/7515)
Launch gdb with
  gdb /home/ehsan/.rr/firefox-1/mmap_1_hardlink_firefox
and attach to the rr debug server with:
  target remote :16914
```

This is on the latest master.

**Comment by rocallahan** [comment]
This is probably the same as #1603.

**Comment by rocallahan** [comment]
@dbaron did a system update that made this go away. So I think I need access to your system to debug this.

**Comment by rocallahan** [comment]
Hopefully fixed by https://github.com/mozilla/rr/commit/75859fd3c0087731550bdba788880f5a7a393c7d. Please update to master and retry.

**Comment by ehsan** [comment]
This still happens for me with 75859fd.

**Comment by rocallahan** [comment]
Fixed by ef66f71a6331dff485ab4fbd97254fc172db5662.
